config/index.js

```diff
@@ -0,0 +1,179 @@
+          console.error('Unable to get template');
+          console.error(err);
+        });
+  });
+}
+
+/**
+ * Print the last 5 available Firebase Remote Config template metadata from the server.
+ */
+function listVersions() {
+  getAccessToken().then(function(accessToken) {
+    const options = {
+      uri: HOST + PATH + ':listVersions?pageSize=5',
+      method: 'GET',
+      resolveWithFullResponse: true,
+      headers: {
+        'Authorization': 'Bearer ' + accessToken,
+      },
+    };
+
+    rp(options)
+        .then(function(resp) {
+          console.log('Versions:');
+          console.log(resp.body);
+        })
+        .catch(function(err) {
+          console.error('Unable to list versions');
+          console.error(err);
+        });
+  });
+}
+
+/**
+ * Roll back to an available version of Firebase Remote Config template.
+ *
+ * @param version Version of the template to roll back to.
+ */
+function rollback(version) {
+  getAccessToken().then(function(accessToken) {
+    const options = {
+      uri: HOST + PATH + ':rollback',
+      method: 'POST',
+      gzip: true,
+      json: true,
+      resolveWithFullResponse: true,
+      headers: {
+        'Authorization': 'Bearer ' + accessToken,
+        'Content-Type': 'application/json; UTF-8',
+        'Accept-Encoding': 'gzip',
+      },
+      body: {
+        version_number: version
+      }
+    };
+
+    rp(options)
+        .then(function(resp) {
+          console.log('Rolled back to: ' + version);
+          console.log(resp.body);
+          const newETag = resp.headers['etag'];
+          console.log('ETag from server: ' + newETag);
+        })
+        .catch(function(err) {
+          console.error('Request to roll back to template: ' + version + ' failed.');
+          console.error(err);
+        });
+  });
+}
+
+/**
+ * Publish the local template stored in `config.json` to the server.
+ *
+ * @param {String} etag ETag must be supplied when publishing a template. This is to
+ *                      avoid race conditions when publishing.
+ */
+function publishTemplate(etag) {
+  getAccessToken().then(function(accessToken) {
+    var options = {
+      method: 'PUT',
+      uri: HOST + PATH,
+      body: fs.readFileSync(`remote-config-${ENVIRONMENT}.json`, 'UTF8'),
+      gzip: true,
+      resolveWithFullResponse: true,
+      headers: {
+        'Authorization': 'Bearer ' + accessToken,
+        'Content-Type': 'application/json; UTF-8',
+        'If-Match': etag
+      }
+    };
+    rp(options)
+        .then(function(resp) {
+          var newETag = resp.headers['etag'];
+          console.log('Template has been published');
+          console.log('ETag from server: ' + newETag);
+        })
+        .catch(function(err) {
+          console.error('Unable to publish template.');
+          console.error(err);
+        });
+  });
+}
+
+const action = process.argv[2];
+const etagOrVersion = process.argv[3];
+
+if (action && action === 'get') {
```

Comment:
Looks a tad messy.  Is it possible to use a switch instead (if the answer isn't obvious in ten seconds to an experienced .js hand as you it probably isn't easy and the existing solution should be used instead)

Reply:
Switch is absolutely possible, but ref earlier comment, this is based on a [template from firebase](https://github.com/firebase/quickstart-nodejs/tree/master/config), so I didn't want to modify it too much.